core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -324,11 +324,20 @@ object LogConfig {
   }
 
   /**
+    * Check that the property values are valid relative to each other
+    */
+  def validateValues(props: Properties) {
+    require(props.getProperty(SegmentBytesProp).toLong <= props.getProperty(RetentionBytesProp).toLong,
```

Comment:
We should probably throw `ConfigException` for consistency with other validation exceptions (Config supports validators and that's the exception that they throw). A bit odd that `validateNames` throws `InvalidConfigurationException`, but it seems like the callers just catch all exceptions.